content/articles/2017-11-15-withr-2.1.0.Rmd

```diff
@@ -0,0 +1,226 @@
+---
+title: withr 2.1.0
+date: '2017-11-15'
+slug: withr-2.1.0
+author: Jim Hester
+categories: [package]
+description: >
+  withr 2.1.0 is now available on CRAN.
+photo:
+  url: https://unsplash.com/photos/V1YEEItoOTE
+  author: Suganth
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  comment = "#>",
+  collapse = TRUE
+)
+library(withr)
+```
+
+[withr 2.1.0](http://withr.tidyverse.org/news/index.html) is now available on CRAN!
+[withr](http://withr.r-lib.org) makes working with global state in R safer and
+less error prone. It also has no non-base dependencies so is easy to include in
+packages.
+
+Install the latest version with:
+```{r, eval = FALSE}
+install.packages("withr")
+```
+
+# Whither withr?
+
+Many functions in R modify global state in some fashion. Some common examples
+are `par()` for graphics parameters, `dir()` to change the current directory
+and `options()` to set a global option. Using these global options is useful
+when using R interactively, because you can set them early in your
+experimentation and they will remain set for the duration of the session.
+However this makes programming with these settings difficult, because in a
+pure function you should always reset the previous state when the function
+completes if you change a global setting.
+
+One common idiom for dealing with this problem is to save the current state,
+make your change, then reset the previous state.
+
+```{r}
+par("col" = "black")
+my_plot <- function(new) {
+  old <- par(col = "red", pch = 19)
+  plot(mtcars$hp, mtcars$wt)
+  par(old)
+}
+my_plot()
+par("col")
+```
+
+However one issue when there is an error before you are able to reset the
+options.
```

Comment:
"However this approach can fail if there's an error before you are able to reset the options."

Reply:
Thanks! I think this frankenstein sentence was the result of a revision gone bad.